pkg/sbom/telemetry/telemetry.go

```diff
@@ -82,3 +85,120 @@ var (
 		commonOpts,
 	)
 )
+
+// QueueMetricProvider is a workqueue.MetricsProvider that provides metrics for the SBOM queue
+type QueueMetricProvider struct{}
+
+type gaugeWrapper struct {
+	telemetry.Gauge
+}
+
+// Inc implements the workqueue.GaugeMetric interface
+func (g gaugeWrapper) Inc() {
+	g.Gauge.Inc()
+}
+
+// Dec implements the workqueue.GaugeMetric interface
+func (g gaugeWrapper) Dec() {
+	g.Gauge.Dec()
+}
+
+// Set implements the workqueue.GaugeMetric interface
+func (g gaugeWrapper) Set(v float64) {
+	g.Gauge.Set(v)
+}
+
+type counterWrapper struct {
+	telemetry.Counter
+}
+
+// Inc implements the workqueue.CounterMetric interface
+func (c counterWrapper) Inc() {
+	c.Counter.Inc()
```

Comment:
The workqueue accepts only `Inc()` and our counter implements `Int(tags ...string)`